vllm/utils.py

```diff
@@ -679,11 +679,18 @@ def make_tensor_with_pad(
     np_dtype = TORCH_DTYPE_TO_NUMPY_DTYPE[dtype]
     padded_x = make_ndarray_with_pad(x, pad, np_dtype, max_len=max_len)
 
-    tensor = torch.from_numpy(padded_x).to(device)
-    if pin_memory:
-        tensor = tensor.pin_memory()
+    return async_numpy_to_tensor(padded_x, device)
 
-    return tensor
+
+def async_numpy_to_tensor(x: npt.NDArray, device: Union[str, torch.device]):
+    """
+    Make a tensor from a numpy array asynchronously. Use pinned memory
+    if possible.
+    """
+    t = torch.from_numpy(x)
+    if is_pin_memory_available():
+        t = t.pin_memory()
+    return t.to(device, non_blocking=True)
```

Comment:
This is even slower than just .to()

https://pytorch.org/tutorials/intermediate/pinmem_nonblock.html

Reply:
I reverted the changes in utils because they are not relavent to this PR anymore.